.github/workflows/validate.yml

```diff
@@ -113,7 +113,7 @@ jobs:
         id: setup-haskell
         with:
           ghc-version: ${{ matrix.ghc }}
-          cabal-version: latest # latest is mandatory for cabal-testsuite, see https://github.com/haskell/cabal/issues/8133
+          cabal-version: 3.12.1.0 # see #10251
```

Comment:
I have a slight preference to full URLs but that's debatable, I guess...

Reply:
Changed to full URLs.